libpod/runtime_pod_linux.go

```diff
@@ -50,6 +50,16 @@ func (r *Runtime) NewPod(ctx context.Context, options ...PodCreateOption) (_ *Po
 		pod.config.Name = name
 	}
 
+	if (pod.config.InfraContainer.CPUPeriod == 0 || pod.config.InfraContainer.CPUQuota == 0) && (pod.config.CPUPeriod != 0 && pod.config.CPUQuota != 0) {
+		pod.config.InfraContainer.CPUPeriod, pod.config.InfraContainer.CPUQuota = pod.config.CPUPeriod, pod.config.CPUQuota
+	}
+
+	pod.config.InfraContainer.ResourceLimits = &spec.LinuxResources{}
+	pod.config.InfraContainer.ResourceLimits.CPU = &spec.LinuxCPU{}
+	if pod.config.InfraContainer.ResourceLimits != nil && pod.config.InfraContainer.ResourceLimits.CPU.Cpus == "" && pod.config.ResourceLimits != nil && pod.config.ResourceLimits.CPU.Cpus != "" {
```

Comment:
You just set the two fields above so the checks in this line are not necessary.

Just set the value.

Reply:
Sorry, that was leftover from testing what was causing nil pointer error b/c ResourceLimits and ResourceLimits.CPU are both pointers.

Reply:
> You just set the two fields above so the checks in this line are not necessary.
> 
> Just set the value.

changing it to `if p.config.InfraContainer.ResourceLimits != nil && p.config.ResourceLimits.CPU != nil` since it you do a simple `podman pod create` ResourceLimits.CPU (I think) will be a nil pointer to a nil pointer